sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/XmlSensorTest.java

```diff
@@ -76,6 +77,38 @@
 
   private final RuleKey parsingErrorCheckRuleKey = RuleKey.of(CheckRepository.REPOSITORY_KEY, parsingErrorCheckKey);
 
+  @Test
+  public void testPerformance() throws Exception {
+    init(false, temporaryFolder.getRoot());
+    File smallFile = temporaryFolder.newFile("smallFile.xml");
+    File bigFile = temporaryFolder.newFile("bigFile.xml");
+    String simpleTag = "<tag1 attr=\"val1\">text</tag1>\n";
+    String xml = "<?xml version=\"1.0\"?><root>\n";
+    StringBuilder str = new StringBuilder(xml);
+    for (int i = 0; i < 20000; i++) {
+      str.append(simpleTag);
+    }
+    str.append("</root>");
+    FileUtils.write(smallFile, str.toString());
+    fs.add(createInputFile(smallFile.getAbsolutePath()));
+    long t1 = System.currentTimeMillis();
+    sensor.analyse(context);
+    long diffSmall = System.currentTimeMillis() - t1;
+
+    str = new StringBuilder(xml);
+    for (int i = 0; i < 40000; i++) {
+      str.append(simpleTag);
+    }
+    str.append("</root>");
+    FileUtils.write(bigFile, str.toString());
+    init(false, temporaryFolder.getRoot());
+    fs.add(createInputFile(bigFile.getAbsolutePath()));
+    t1 = System.currentTimeMillis();
+    sensor.analyse(context);
+    long diffBig = System.currentTimeMillis() - t1;
+    assert (diffBig < (2.5 * diffSmall));
```

Comment:
This is not a valid assertions for tests. We don't use `assert` statement but assertions from testing frameworks. Look at the other tests in this file, and use Assertions from `assertj` using `assertThat()` the same way it is used for instance in sonar-java performance testing: https://github.com/SonarSource/sonar-java/blob/master/its/performancing/src/test/java/JavaPerformanceTest.java